config/stored_requests_test.go

```diff
@@ -154,6 +157,100 @@ func TestInMemoryCacheValidationSingleCache(t *testing.T) {
 	}).validate(AccountDataType, nil))
 }
 
+func TestPostgresConfigValidation(t *testing.T) {
+	tests := []struct {
+		description            string
+		connectionStr          string
+		cacheInitQuery         string
+		cacheInitTimeout       int
+		cacheUpdateQuery       string
+		cacheUpdateRefreshRate int
+		cacheUpdateTimeout     int
+		existingErrors         []error
+		wantErrorCount         int
+	}{
+		{
+			description:   "No connection string",
+			connectionStr: "",
+		},
+		{
+			description:   "Connection string but no queries",
+			connectionStr: "some-connection-string",
+		},
+		{
+			description:    "Valid cache init query with zero timeout",
+			connectionStr:  "some-connection-string",
+			cacheInitQuery: "SELECT * FROM table;",
+			wantErrorCount: 1,
+		},
+		{
+			description:      "Valid cache init query with non-zero timeout",
+			connectionStr:    "some-connection-string",
+			cacheInitQuery:   "SELECT * FROM table;",
+			cacheInitTimeout: 1,
+		},
+		{
+			description:            "Valid cache update query with non-zero timeout and refresh rate",
+			connectionStr:          "some-connection-string",
+			cacheUpdateQuery:       "SELECT * FROM table WHERE $1",
+			cacheUpdateRefreshRate: 1,
+			cacheUpdateTimeout:     1,
+		},
+		{
+			description:            "Valid cache update query with zero timeout and non-zero refresh rate",
+			connectionStr:          "some-connection-string",
+			cacheUpdateQuery:       "SELECT * FROM table WHERE $1",
+			cacheUpdateRefreshRate: 1,
+			cacheUpdateTimeout:     0,
+			wantErrorCount:         1,
+		},
+		{
+			description:            "Valid cache update query with non-zero timeout and zero refresh rate",
+			connectionStr:          "some-connection-string",
+			cacheUpdateQuery:       "SELECT * FROM table WHERE $1",
+			cacheUpdateRefreshRate: 0,
+			cacheUpdateTimeout:     1,
+			wantErrorCount:         1,
+		},
+		{
+			description:            "Invalid cache update query missing wildcard",
+			connectionStr:          "some-connection-string",
+			cacheUpdateQuery:       "SELECT * FROM table",
+			cacheUpdateRefreshRate: 1,
+			cacheUpdateTimeout:     1,
+			wantErrorCount:         1,
+		},
+		{
+			description:      "Multiple errors: valid queries missing timeouts and refresh rates plus existing error",
+			connectionStr:    "some-connection-string",
+			cacheInitQuery:   "SELECT * FROM table;",
+			cacheUpdateQuery: "SELECT * FROM table WHERE $1",
+			existingErrors:   []error{errors.New("existing error before calling validate")},
+			wantErrorCount:   4,
+		},
```

Comment:
Can you please add test coverage for:
```
	if strings.Contains(cfg.Query, "$") {
		errs = append(errs, fmt.Errorf("%s: postgres.initialize_caches.query should not contain any wildcards (e.g. $1)", section))
	}
```

That looks to me as the only missing check.